java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/menu/handlers/DeleteCategoryHandler.java

```diff
@@ -0,0 +1,67 @@
+package com.xxAMIDOxx.xxSTACKSxx.menu.handlers;
+
+import com.xxAMIDOxx.xxSTACKSxx.core.messaging.publish.ApplicationEventPublisher;
+import com.xxAMIDOxx.xxSTACKSxx.menu.commands.DeleteCategoryCommand;
+import com.xxAMIDOxx.xxSTACKSxx.menu.domain.Category;
+import com.xxAMIDOxx.xxSTACKSxx.menu.domain.Menu;
+import com.xxAMIDOxx.xxSTACKSxx.menu.events.CategoryDeletedEvent;
+import com.xxAMIDOxx.xxSTACKSxx.menu.events.MenuEvent;
+import com.xxAMIDOxx.xxSTACKSxx.menu.events.MenuUpdatedEvent;
+import com.xxAMIDOxx.xxSTACKSxx.menu.exception.CategoryDoesNotExistException;
+import com.xxAMIDOxx.xxSTACKSxx.menu.exception.ItemAlreadyExistsException;
+import com.xxAMIDOxx.xxSTACKSxx.menu.repository.MenuRepository;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * @author ArathyKrishna
+ */
+@Component
+public class DeleteCategoryHandler extends MenuBaseCommandHandler<DeleteCategoryCommand> {
+
+    public DeleteCategoryHandler(MenuRepository menuRepository,
+                                 ApplicationEventPublisher applicationEventPublisher) {
+        super(menuRepository, applicationEventPublisher);
+    }
+
+    @Override
+    Optional<UUID> handleCommand(Menu menu, DeleteCategoryCommand command) {
+        Category category = getCategory(menu, command);
+        if (!category.getItems().isEmpty()) {
+            throw new ItemAlreadyExistsException(command, command.getCategoryId(), "");
+        }
+        List<Category> collect = menu.getCategories().stream()
+                .filter(t -> !Objects.equals(t, category))
+                .collect(Collectors.toList());
+        menu.setCategories(!collect.isEmpty() ? collect : null);
+        menuRepository.save(menu);
+        return Optional.empty();
+    }
+
+    @Override
+    List<MenuEvent> raiseApplicationEvents(Menu menu,
+                                           DeleteCategoryCommand command) {
+        return Arrays.asList(new MenuUpdatedEvent(command),
+                new CategoryDeletedEvent(command, command.getCategoryId()));
+    }
+
+    Category getCategory(Menu menu, DeleteCategoryCommand command) {
```

Comment:
You're likely doing the same check in UpdateCategory as DeleteCategory - try to avoid duplicating code